contracts/Zauction.sol

```diff
@@ -35,24 +37,33 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     uint256 amount,
     address nftAddress,
     uint256 tokenId,
-    uint256 expireBlock
+    uint256 expireBlock,
+    IERC20 paymentToken
   );
 
   event DomainSold(
     address indexed buyer,
     address indexed seller,
     uint256 amount,
     address nftAddress,
-    uint256 indexed tokenId
+    uint256 indexed tokenId,
+    IERC20 paymentToken
```

Comment:
Instead of emitting all of the IERC20, I wonder if we should use the IERC20 functions to emit the relevant data fields instead of the entire IERC20 type. 

Namely:
`paymentTokenName: name()`
`paymentTokenDecimals: decimals()`

We should also consider emitting the `paymentTokenAddress`  and the `topLevelDomainId` (network).

I see these fields being useful for computing data for questions like:
How many domains have been sold on a specific network?
What tokens are being used to sell domains?
What percentage of domain sales is a specific token responsible for?

Reply:
For clarity when reading it we can just have it be the address but from your end consuming it, it would be the same either way. When it's emitted it would just be the address. 

According to the [EIP on OpenZeppelin](https://eips.ethereum.org/EIPS/eip-20) things like `name` and `decimals are optional and should not be expected to be present in smart contracts. You can look at the [IERC20 interface code](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/IERC20.sol) on and see they are not there as well. 

Good idea on the top level ID. That can be gotten through the contract but adding in the events means we can avoid having to make that call in future consuming code. Will add